plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -174,10 +174,16 @@
      * The OpenShift deploy timeout in seconds:
      * See this issue for background of why for end users on slow wifi on their laptops
      * DeploymentConfigs usually barf: https://github.com/openshift/origin/issues/10531
+     *
+     * Lets default to 2 hours which should be plenty of time on a slow connection to download
+     * all the images for a YAML/JSON manifest (which can contain many separate images)
      */
-    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds")
+    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds", defaultValue = "7200")
```

Comment:
It doesn't matter whether you use `mvn fabric8:deploy`, `oc apply`, `gofabric8 deploy` or instantiate a Template via the openshift console - in all cases openshift uses the timeout from the YAML to determine how long all the `DeploymentConfig`s have before they all load the docker images + startup. Using the OOTB 10 minutes timeout means I'm unable to ever deploy anything on minishift with > 3 or 4 images inside. 

I totally understand that this is really openshift's issue; it should differentiate between time taken to download images (which when running a YAML with many images inside can take a while for them all to download) versus time taken to really startup the pods.

But the question is faced with the various openshift releases out there right now (OSCP/OSD/OSO/minishift) what should app developersput in their released YAMLs to make them as useful as they can for end users consuming them via oc/openshift console/gofabric8/mvn.